twine/repository.py

```diff
@@ -88,12 +102,15 @@ def upload(self, package):
                 (package.basefilename, fp, "application/octet-stream"),
             ))
             encoder = MultipartEncoder(data_to_send)
+            monitor = MultipartEncoderMonitor(
```

Comment:
So if `upload_callback` returned `callback` and `bar`, you could call `bar` after L114